### DISCUSSION TITLE
How to apply a regular expression to use only a part of a URL as a title.

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#47] [Q&A] [answered]
<img width="658" alt="image" src="https://user-images.githubusercontent.com/67494083/149212281-59bc31d6-301c-4e0d-81b4-27b79c653ec4.png">

As mentioned in the documentation, can you provide an example of using a regular expression to use only a part of the URL as title of the clipped tiddler.

**Comment by buggyj** [comment, marked as answer]
As a example we can remove the 'path' part of a url, eg this page has url
`https://github.com/buggyj/tiddlyclip-plugin/discussions/47`

we can remove upto the last `/`  can just use the number i.e '47' as part of the tiddler title, combining it with 'discussion' i.e. 

"discussion 47"

----

step 1 create a new row in the TiddlyClipConfig

![image](https://user-images.githubusercontent.com/2855795/149305728-ad7cd9b3-509c-4737-b2cd-9d2910450945.png)

----

step 2 clone the defaultSnip tiddler and rename it DiscussionsSnip, then change the 'Title' to refrence another tiddler called 'TitleRule'

![image](https://user-images.githubusercontent.com/2855795/149306466-0b31f5a0-8cd1-4c17-bb5f-ee353a22fc95.png)

----

create the 'TitleRule' Tiddler, make sure the 'type is as shown'

![image](https://user-images.githubusercontent.com/2855795/149306880-920343c3-09fb-4d11-804c-037b67b13709.png)


-----

reload you tiddlywiki and re-dock so the new method appears in the browser tiddlyclip context menu.

snip some text from this webpage (this discussion) using new method.

see the result -

![image](https://user-images.githubusercontent.com/2855795/149307621-69a4b5ee-6eff-42c1-951d-d2be5a94c744.png)

**Reply by arunnbabu81** [reply]
I tried, but it's not working. I have attached a screenshot

<img width="1806" alt="image" src="https://user-images.githubusercontent.com/67494083/149389771-59920eb4-bddf-458e-ba04-3ea933715c7f.png">

What might be going wrong?

**Reply by buggyj** [reply]
the contents of the TitleRule should be

`#rule=/[\s\S]*\/(.*?)$/g`
`#term=$1`
`$title=Discussion ((*@pageRef/#rule/#term*))`

**Reply by arunnbabu81** [reply]
Thank you. I mistook `\/` as `V`. Since I don't have any programming background I couldn't recognise that mistake.

Now its working
